[FEATURE] Add time division props to clock

arthur5005 opened the pull request:
* This addresses #75 and includes new time divisions props on the Clock util, and adds time division props to clock service as `@tracked` properties instead of just getter passthroughs so that the `@computed` macro can react to them

tzellman commented:
Thanks for the PR!! I will review and do some testing locally and hopefully get it merged very soon.

arthur5005 commented on addon/services/clock.js:
```diff
@@ -30,8 +37,16 @@ export default class ClockService extends Service {
     }
 
+    updateTrackedDivisions() {
+        this.second = this.date.getSeconds();
+        this.minute = this.date.getMinutes();
+        this.hour = this.date.getHours();
+        this.day = this.date.getDate();
```
Yeah, good call, will change.

arthur5005 commented:
@tzellman
I realized I made a silly mistake by updating every tracked property on every tick, forgot that Ember changed the way it does dirty tracking. 

I've done the following
* I've updated the PR with the proper guards in place
* Updated the README.md with a clear example of how to use the tracked properties
* Updated the dummy app, to effectively include the example from the README.md
* Added an integration test that _actually_ tests Ember's rendering / refreshing behaviour.

Effectively, as it always is, I got bit by not having enough tests!

tzellman commented on tests/dummy/app/components/clock-demo.js:
```diff
@@ -1,6 +1,22 @@
+
+    @computed('clock.minute')
+    get timeByTheMinute() {
+        return this.clock.time;
```
Should be `this.clock.minute`? Also, the other two computed properties may need to be updated as well. Thanks so much for your PR!

arthur5005 replied:
Hmm no, not really. This is trying to illustrate that the `timeByMinute` only updates the template with a new value every minute as a result of the computed macro.

Perhaps a better example would be if I did:

```javascript
    @computed('clock.minute')
    get timeByTheMinute() {
        return (new Date()).getTime();
    }
```

So that readers don't think there's a mistake or a different dependency.

arthur5005 replied:
I've gone ahead and updated the PR again. 

And no problem, I'm just glad there's a more modern version of a package like this, and I'm keen to get it to be a drop in replacement for https://github.com/lozjackson/ember-clock

:), certainly helps our project out.

tzellman commented:
Thanks for the PR!
